### PR TITLE
Change magic intrinsics

### PR DESCRIPTION
**Pull request by jez**
Review by commit for better diffs.

### Motivation

Hygiene.

### Test plan

See included automated tests.

### CODE REVIEW

File: core/types/calls.cc [hunk 1]
```diff
@@ -1999,14 +1999,13 @@ const vector<Intrinsic> intrinsicMethods{
...
 
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::buildHash(), &Magic_buildHash},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::buildArray(), &Magic_buildArray},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::expandSplat(), &Magic_expandSplat},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::callWithSplat(), &Magic_callWithSplat},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::callWithBlock(), &Magic_callWithBlock},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::callWithSplatAndBlock(),
-     &Magic_callWithSplatAndBlock},
-    {Symbols::MagicSingleton(), Intrinsic::Kind::Instance, Names::suggestType(), &Magic_suggestUntypedConstantType},
+    {Symbols::Magic(), Intrinsic::Kind::Singleton, Names::buildHash(), &Magic_buildHash},
```
Comment (DarkDimius) [inline core/types/calls.cc]: What's the modification behind the change? Why do you want to call this on an instance of `Magic`?

Comment (jez) [reply core/types/calls.cc]: It's just easier to read. The `Magic.<foo>` methods are like the `T.foo` methods.

But for the `Magic` methods, we manually got our hands on the `MagicSingleton` class, and then register the intrinsics as instance methods. but that's exactly what `Intrinsic::Kind::Singleton` does, and was what we were already using the for the `T` methods.

Comment (DarkDimius) [reply core/types/calls.cc]: Ack, makes sense!

**Comment by DarkDimius** [comment]
Do we still need `MagicSingleton` at all after this?

**Comment by jez** [comment]
> Do we still need `MagicSingleton` at all after this?

Yes, because we need to enter the methods themselves in GlobalState. For the `T` methods, since the names are valid Ruby syntax, we can define their arity in RBI files. But for these methods, GlobalState is the only option.

**Pull request merged**